internal/services/groups/groups_data_source.go

```diff
@@ -59,6 +59,12 @@ func groupsDataSource() *schema.Resource {
 				Optional:     true,
 				ExactlyOneOf: []string{"display_names", "object_ids", "return_all"},
 			},
+
+			"security_enabled": {
+				Description: "Retrieve only groups that are security_enabled groups",
+				Type:        schema.TypeBool,
+				Optional:    true,
```

Comment:
I think this should be mutually exclusive with the `return_all` property?

```suggestion
				Optional:    true,
				ConflictsWith: []string{"return_all"},
```

Reply:
I think it should be possible to query for all groups that are security_enabled ?
 
I know it is a complication with regards to logic but the ability to just filter out 365 groups could be a useful tool.

Reply:
Yeah the more I've thought about it, I agree it should be possible to specify `return_all` and filter out mail and security groups.

Reply:
Filter-out or filter in?

Reply:
Sorry filter in, as in I should have written "filter for" :D

Reply:
I suppose what we should think about aswell is the oppertunity for future work - Perhaps people will want to specify `mail_enabled=false` and only get non-mail_enabled groups

Reply:
Right, that's sort of what I was trying to get at with the inclusion of both properties. Someone may want to specify `security_enabled=true` and `mail_enabled=false` to get security groups that aren't mail enabled (e.g. effectively only get classic security groups and not potential M365 groups)

Reply:
(You should be able to use d.GetOk() to test if the property was specified versus just having it's default (unset) value)